packages/babel-types/src/definitions/core.ts

```diff
@@ -2157,6 +2206,49 @@ defineType("ClassPrivateProperty", {
   },
 });
 
+defineType("ClassPrivateAccessorProperty", {
```

Comment:
I think that if we do not, we probably need to add some level of validation. There are a lot of extra properties, such as `computed`, on the public property which do not apply to the private one. 

I think my biggest worry is that this will result in a very different API than the norms that were established by `ClassPrivateProperty` and `ClassPrivateMethod`. Users of plugins will likely expect other class elements to follow that convention, and there will probably be some amount of code that has to be somewhat different in order to support it.

I guess I think it would make the most sense if all of these were valid:

```js
class Foo {
  accessor a { get; set; }
  accessor #a { get; set; }
  accessor a { #get; #set; }
  accessor #a { #get; #set; }
}
```

But I think the last case there is invalid, if it's already private then it can't be made _more_ private, so I think the distinction can still be useful and is keeping with the convention.

Reply:
> accessor a { #get; #set; }

`#get` is not proposed so `#get` is invalid.

> accessor #a { #get; #set; }

True. This is invalid because only accessor properties with Identifier names can have a #set stub or #set method.

The case `accessor a { get; #set; }` pushes me to merge the two, because we still need to handle the private elements in an `ClassAccessorProperty`. For example we can think of the transformer routine:

```js
if (path.isClassAccessorProperty()) {
  if (path.get("set").isPrivate()) {
    // handle private
  }
  // handle public
} else if (path.isClassPrivateAccessorProperty()) {
  // handle private
}
```
versus
```js
if (path.isClassAccessorProperty()) {
  if (path.get("key").isPrivate || path.get("set").isPrivate()) {
    // handle private
  }
  // handle public
}
```

Reply:
I think it's pretty likely that the transforms for `accessor a { get; #set; }` will be fairly different from `accessor #a { get; set; }`, so I'm not sure if it will end up being that much of a benefit, but if it makes more sense to represent them with the same node type then we can do that.